dashboard/app/controllers/hoc_download_controller.rb

```diff
@@ -0,0 +1,42 @@
+class HocDownloadController < ApplicationController
+
+  DOWNLOAD_URL_BASE = '//downloads.code.org/hourofcode/'
+
+  def index
+    @product_name = params[:product]
+
+    case @product_name
+      when Script::MINECRAFT_NAME
+        @app_name = t('hoc_download.minecraft_name')
+        @file_prefix = 'MC'
+        @og_image_url = CDO.code_org_url('/images/mc/mc_social.jpg')
+      when Script::STARWARS_NAME
+        @app_name = t('hoc_download.starwars_blocks_name')
+        @file_prefix = 'StarWarsBlocks'
+        @og_image_url = CDO.code_org_url('/images/star-wars-announcement.jpg')
+
+        # for JavaScript version (Star Wars only)
+        @app_name_js = t('hoc_download.starwars_javascript_name')
+        @file_prefix_js = 'StarWarsJS'
+        @show_js_links = true
+      else
+        render_404
+        return
+    end
+
+    @hoc_url = CDO.code_org_url("/#{@product_name}")
+
+    @english_windows_url = DOWNLOAD_URL_BASE + @product_name + '/' + @file_prefix + "EnglishSetup.exe"
+    @english_mac_url = DOWNLOAD_URL_BASE + @product_name + '/' + @file_prefix + "English.dmg"
+    @spanish_windows_url = DOWNLOAD_URL_BASE + @product_name + '/' + @file_prefix + "SpanishSetup.exe"
+    @spanish_mac_url = DOWNLOAD_URL_BASE + @product_name + '/' + @file_prefix + "Spanish.dmg"
```

Comment:
Yeah, instead of hard coding this path, do something along these lines: https://github.com/code-dot-org/code-dot-org/blob/f7e69d0f9f62f94c207235c2c0d79091cc7d215f/dashboard/app/helpers/application_helper.rb#L89-L89

And dig into [CDO.site_url](https://github.com/code-dot-org/code-dot-org/blob/c4c12dcf307196860270955c379b6b4137224f6c/deployment.rb#L182-L182)

Also nit, use ruby string interpolation, i.e. `@english_mac_url = "#{DOWNLOAD_URL_BASE}#{@product_name}/@{file_prefix}English.dmg"`. Though, you can probably reduce this with the above to something like ` = download_url(@product_name, @file_prefix, 'English.dmg')`

I agree also with Brendan that this should be in a lookup, instead of a bunch of separate variables.